From 847222aac40490cd37826a1409b136206791cb18 Mon Sep 17 00:00:00 2001
From: Kuppuswamy Sathyanarayanan <sathyanarayanan.kuppuswamy@intel.com>
Date: Mon, 22 Sep 2014 15:56:58 -0700
Subject: iio: bmm050: Fix suspend/resume state issue

device op mode setting is not restored properly in resume
call of the device. This patch fixes that problem.

Signed-off-by: Kuppuswamy Sathyanarayanan <sathyanarayanan.kuppuswamy@intel.com>
---
 drivers/iio/magnetometer/bmm050_driver.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/iio/magnetometer/bmm050_driver.c b/drivers/iio/magnetometer/bmm050_driver.c
index c576e5f..f94d526 100644
--- a/drivers/iio/magnetometer/bmm050_driver.c
+++ b/drivers/iio/magnetometer/bmm050_driver.c
@@ -1549,6 +1549,13 @@ static int bmm_resume(struct device *dev)
 	/* And then restore the HW configuration */
 	mutex_lock(&client_data->mutex_power_mode);
 	err = bmm_restore_hw_cfg(client);
+	err = bmm_set_op_mode(client_data, client_data->op_mode);
+	if (err) {
+		dev_err(&client->dev, "fail to set opmode %d",
+						client_data->op_mode);
+		mutex_unlock(&client_data->mutex_power_mode);
+		return err;
+	}
 	/* post resume operation */
 	bmm_post_resume(client);
 
-- 
1.9.1

